src/fcc_bridge_node_ros.cpp

```diff
@@ -0,0 +1,225 @@
+
+    // Setup MAVSDk objects such as system, telemetry etc.
+    this->setup_mavsdk();
+    if (this->internal_state == INTERNAL_STATE::ERROR) {
+        RCLCPP_FATAL(this->get_logger(), "Failed to setup MAVSDK! Exiting...");
+        this->exit_process_on_error();
+    }
+    RCLCPP_INFO(this->get_logger(), "Transitioning into MAVSDK_SET_UP state");
+    this->internal_state = INTERNAL_STATE::MAVSDK_SET_UP;
+
+    // Activating node to signal that it is ready for the safety limits
+    this->activate();
+
+    RCLCPP_INFO(this->get_logger(), "Successfully created %s instance",
+                this->get_name());
+}
+
+void FCCBridgeNode::setup_ros() {
+    RCLCPP_DEBUG(this->get_logger(), "Setting up ROS");
+    // Check if the node is in the correct state
+    if (this->internal_state != INTERNAL_STATE::STARTING_UP) {
+        RCLCPP_ERROR(this->get_logger(),
+                     "Repeated try to setup ros components");
+        this->internal_state = INTERNAL_STATE::ERROR;
+        return;
+    }
+
+    // Setup publisher
+
+    // Create GPSInfo & Position publisher
+    this->gps_position_publisher =
+        this->create_publisher<interfaces::msg::GPSPosition>("uav_gps_position",
+                                                             1);
+    // Create FlightState publisher
+    this->flight_state_publisher =
+        this->create_publisher<interfaces::msg::FlightState>("uav_flight_state",
+                                                             1);
+
+    // Create BatteryState publisher
+    this->battery_state_publisher =
+        this->create_publisher<interfaces::msg::BatteryState>(
+            "uav_battery_state", 1);
+
+    // Create RCState publisher
+    this->rc_state_publisher =
+        this->create_publisher<interfaces::msg::RCState>("uav_rc_state", 1);
+
+    // Setup subscriber
+
+    // Create subscription options for heartbeat to only receive mission control
+    // heartbeats
+    rclcpp::SubscriptionOptions subscription_options;
+    subscription_options.content_filter_options.filter_expression =
+        "sender_id = 'mission_control'";
+
+    // Create Heartbeat subscription
+    this->mission_control_heartbeat_subscriber =
+        this->create_subscription<interfaces::msg::Heartbeat>(
+            "heartbeat", 1,
+            std::bind(&FCCBridgeNode::mission_control_heartbeat_subscriber_cb,
+                      this, std::placeholders::_1),
+            subscription_options);
+
+    // Ensure that the filter is enabled. (Some DDS versions do not support it)
+    if (!this->mission_control_heartbeat_subscriber->is_cft_enabled()) {
+        RCLCPP_FATAL(this->get_logger(), "Content filtering is not enabled!");
+        this->internal_state = INTERNAL_STATE::ERROR;
+        return;
+    }
+
+    // Setup 5Hz timer to get telemetry from the FCC
+    this->fcc_telemetry_timer_5hz = this->create_wall_timer(
+        std::chrono::milliseconds{200},

```

Comment:
Maybe think about outsourcing the values into constants for easier maintainability. (same for the 10 Hz)

Reply:
Fixed in 185a86c
Da war auch noch ein ganz böser Copy paste error.